src/Architecture.cpp

```diff
@@ -185,20 +189,43 @@ Architecture::Architecture(const std::uint16_t nQ, const CouplingMap& cm,
 }
 
 void Architecture::createDistanceTable() {
+  Matrix<double> edgeWeights(nqubits, std::vector<double>(nqubits, INFINITY));
```

Comment:
Prefer using the appropriate `max` value from `std::numeric_limits` instead of INFINITY.